src/Features/Core/Portable/PullMemberUp/MemberAndDestinationValidator.cs

```diff
@@ -0,0 +1,50 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.  
+
+using System.Linq;
+using System.Threading;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.PullMemberUp
+{
+    internal static class MemberAndDestinationValidator
+    {
+        internal static bool IsDestinationValid(Solution solution, INamedTypeSymbol destination, CancellationToken cancellationToken)
+        {
+            if (destination == null)
+            {
+                throw ExceptionUtilities.Unreachable;
```

Comment:
nit: this sort of checking isn't super helpful.  We're effectively saying "this should never happen".  but we get that anyways just by the fact that the next line will immediately crash on `destination.TypeKind`.  So this really isn't necessary.